### DISCUSSION TITLE
up-hungry doesn't update if not in the same overlay

### DISCUSSION THREAD

**Discussion by GabrielNBDS** [unpoly/unpoly#483] [Q&A] [answered]
I'm using `up-hungry` to show notifications and after successfully submitting a form I show a notification.

When successfully submitted the user is redirected and in the new page the notification is shown but when I used `up-layer="new"` to open the form in a modal it stopped working. Using `up-if-layer="any"` makes it possible to show notifications while the modal is opened but still doesn't work if the modal is closed and I want to show the notification afterwards.

Is there anything I can do or is it the expected behavior?

**Comment by adam12** [comment]
First off, can you confirm which version of Unpoly you're using?

When you say modal is closed, was it [accepted? or dismissed](https://unpoly.com/closing-overlays) through a request? I believe it should work as you expect but I think we need more information.

If you can, please enable logs with `up.log.enable()` and share the relevant pieces here as well.

**Comment by GabrielNBDS** [comment]
I'm using unpoly 3.1.0.

It was accepted with `up-accept-location`.

Logs with the modal opened: https://vimeo.com/824353570?share=copy
Logs in page: https://vimeo.com/824354375?share=copy

**Comment by adam12** [comment]
The videos were helpful. The simplest solution right now is likely to force the layer on the `<form>`, by adding an attribute like this: `up-layer="parent or root"`. This will negate the use of `[up-accept-location]` (I believe).

Whether the hungry element should have been uplifted from the response and added to the current page (via `[if-up-layer="any"]` or another mechanism), I am not sure as I haven't used that feature yet. Perhaps @triskweline knows when he has a free moment.

**Comment by GabrielNBDS** [comment]
Thanks. I'm not sure I understood. Should I add `up-layer="parent"` in the modal's form? I tried it and it still didn't work 😬

**Reply by adam12** [reply, marked as answer]
You'd want `up-layer="parent or root"`. See my [Glitch here](https://glitch.com/edit/#!/almondine-western-lifeboat?path=two.html%3A24%3A18).

**Reply by GabrielNBDS** [reply]
It worked! Thanks! I didn't know about that.

**Comment by GabrielNBDS** [comment]
That worked but when there's a server error it closes the modal. I tried adding `up-fail-layer="current"` but it didn't work.

In the log with `up-layer="parent or root"` it shows it was dismissed https://streamable.com/uk5zsw
Without it updates the modal https://streamable.com/3qo4oe

Thanks again for the help so far. I'm new to unpoly

**Comment by adam12** [comment]
I see 200 HTTP response in the log for the one video you shared. Make sure that you're [returning a 4xx status code if the action wasn't successful.](https://unpoly.com/validation#signaling-a-failed-submission)

**Reply by GabrielNBDS** [reply]
that was it, thanks again!
